app/components/goal-tile.jsx

```diff
@@ -2,56 +2,110 @@ import React from 'react';
 
 import * as steps from './../steps.js';
 
-const tile = ({ goal, step, onSelectGoal, }) => {
+const tile = ({ goal, step, onSelectGoal, visibleEditDelete, onBorderClick,
+onDeleteGoal, onEditGoal, }) => {
+
+  const visible = (goal.deleted) ?
+  { display: 'none', } :
+  { display: 'flex', };
+
+  const border = (step === steps.GOALS_LIST) ?
+  { visibility: 'visible', } : { visibility: 'hidden', };
 
   const pathAvatar = avatarName => `./images/avatars/${avatarName}.svg`;
 
   const score = step === steps.RATE_GOAL || step === steps.FEEDBACK
-    ? goal.newRating.score || 0
-    : goal.ratings && goal.ratings.length
-      ? goal.ratings[0].score
-      : undefined;
+  ? goal.newRating.score || 0
+  : goal.ratings && goal.ratings.length
+  ? goal.ratings[0].score
+  : undefined;
 
   const progressStyle = {
     width: score === undefined
-      ? 0
-      : `${score * 10}%`,
+    ? 0
+    : `${score * 10}%`,
     transition: 'width 1s ease',
   };
 
+  const editIcon = './images/icons/edit-icon.svg';
+  const deleteIcon = './images/icons/delete-icon.svg';
+
+  const editStyle = (visibleEditDelete) ?
+  { visibility : 'visible', } :
+  { visibility : 'hidden' , };
+
+  const deleteStyle = (visibleEditDelete) ?
+  { visibility : 'visible', } :
+  { visibility : 'hidden', };
```

Comment:
dunno if I have missed something, but do these need different styles? Are they not always shown at the same time?